datadog/resource_datadog_dashboard.go

```diff
@@ -4196,6 +4217,37 @@ func getScatterplotRequestSchema() map[string]*schema.Schema {
 		},
 	}
 }
+
+func buildDatadogScatterplotTableRequest(terraformRequest map[string]interface{}) *datadogV1.ScatterplotTableRequest {
+	datadogScatterplotTableRequest := datadogV1.NewScatterplotTableRequest()
+
+	if v, ok := terraformRequest["query"].([]interface{}); ok && len(v) > 0 {
+		queries := make([]datadogV1.FormulaAndFunctionQueryDefinition, len(v))
+		for i, q := range v {
+			query := q.(map[string]interface{})
+			if w, ok := query["event_query"].([]interface{}); ok && len(w) > 0 {
+				queries[i] = buildDatadogEventQuery(w[0].(map[string]interface{}))
+			} else if w, ok := query["metric_query"].([]interface{}); ok && len(w) > 0 {
+				queries[i] = buildDatadogMetricQuery(w[0].(map[string]interface{}))
+			} else if w, ok := query["process_query"].([]interface{}); ok && len(w) > 0 {
+				queries[i] = buildDatadogFormulaAndFunctionProcessQuery(w[0].(map[string]interface{}))
+			}
+		}
+		datadogScatterplotTableRequest.SetQueries(queries)
+		datadogScatterplotTableRequest.SetResponseFormat(datadogV1.FormulaAndFunctionResponseFormat("scalar"))
```

Comment:
According to the [docs](https://docs.datadoghq.com/dashboards/widgets/scatter_plot/#api), both `scalar` and `timeseries` values are allowed.

Reply:
While the generated docs show that `scalar` and `timeseries` values are allowed, only `scalar` is allowed for Scatterplot widgets which is why we pin it like this (similar to the geomap, query value, toplist, etc)